pymodbus/transport/transport.py

```diff
@@ -177,16 +177,19 @@ def __init__(
             host = self.comm_params.host
             port = int(self.comm_params.port)
         if self.comm_params.comm_type == CommType.SERIAL:
-            host, port = self.init_setup_serial(host, port)
-            if not host and not port:
+            host_serial, port_serial = self.init_setup_serial(host, port)
+            if host_serial is None or port_serial is None:
                 return
+            host, port = host_serial, port_serial
```

Comment:
This is just complicating things ! please use host, port directly.

Reply:
See my comment 9

Reply:
It is a not acceptable complication, that brings nothing.

Reply:
Well it makes this type correct